src/expandProperty.js

```diff
@@ -244,6 +294,189 @@ function parseGap(value) {
+  // initial value is specified individually for each of the flex-direction and flex-wrap properties
+  let flexDirection = '';
+  let flexWrap = '';
+  const [left, right] = splitShorthand(value)
+
+  const extractFlexFlowValue = (value) => {
+    if (value) {
+      if (flexDirectionValues.has(value)) {
+        if (flexDirection !== '') {
+          // flex-direction has already been set, original value invalid
+          return true;
+        }
+        flexDirection = value;
+      } else if (flexWrapValues.has(value)) {
+        if (flexWrap !== '') {
+          // flex-wrap has already been set, original value invalid
+          return true;
+        }
+        flexWrap = value;
+      } else {
+        // invalid value
+        return true;
+      }
+    }
+
+    return false;
+  }
+
+  if (extractFlexFlowValue(left) || extractFlexFlowValue(right)) {
+    // invalid value, ignore both values
+    return {};
+  }
+
+  return {
+    // only fallback to default value when original value is valid
+    flexDirection: flexDirection || 'row',
+    flexWrap: flexWrap || 'nowrap'
+  }
+}
+
+// https://w3c.github.io/csswg-drafts/css-align/#place-content
+function parsePlaceContent(value) {
+  // The first value is assigned to align-content.
+  // The second value is assigned to justify-content
+  let [alignContent, justifyContent] = splitShorthand(value)
+  if (!justifyContent && alignContent) {
+    // if omitted, ...
+    if (
+      alignContent === 'left'
+      || alignContent === 'right'
+      || alignContent === 'first'
+      || alignContent === 'last'
+    ) {
+      // invalid value, ignore both values
+      return {}
+    }
+    if (alignContent === 'baseline') {
+      // ..., unless that value is a <baseline-position> in which case it is defaulted to start.
+      justifyContent = 'start'
+    } else {
+      // ..., it is copied from the first value
+      justifyContent = alignContent
+    }
+  }
+
+  return {
+    alignContent,
+    justifyContent
+  }
+}
+
+// https://w3c.github.io/csswg-drafts/css-align/#place-items-property
+function parsePlaceItems(value) {
+  // This shorthand property sets both the align-items and justify-items properties in a single declaration. The first value is assigned to align-items. The second value is assigned to justify-items; if omitted, it is copied from the first value.
+  const [alignItems, justifyItems = alignItems] = splitShorthand(value);
+  return {
+    alignItems,
+    justifyItems
+  }
+}
+
+function parsePlaceSelf(value) {
+  // https://w3c.github.io/csswg-drafts/css-align/#place-self-property
+  const [alignSelf, justifySelf = alignSelf] = splitShorthand(value);
+  return {
+    alignSelf,
+    justifySelf
+  }
+}
+
+// https://w3c.github.io/csswg-drafts/css-transitions/#transition-shorthand-property
+// Since transition can accept multiple values, we will run extraction on each value
+const extractTransition = (value) => {
```

Comment:
`extractTransition` is a standalone function.

When `parseTransition` receives a value without `,`, it will return `extractTransition` directly.
When `parseTransition` receives a value that has `,`, it will split the value, then and call `extractTransition` on each one of them.